category_encoders/target_encoder.py

```diff
@@ -113,34 +127,52 @@ def _fit(self, X, y, **kwargs):
 
     def fit_target_encoding(self, X, y):
         mapping = {}
+        prior = self._mean = y.mean()
 
         for switch in self.ordinal_encoder.category_mapping:
             col = switch.get('col')
-            values = switch.get('mapping')
+            if 'HEIR_' not in col:
+                values = switch.get('mapping')
 
-            prior = self._mean = y.mean()
+                scalar = prior
+                if self.heirarchy and col in self.heirarchy:
+                    col_heir = 'HEIR_'+col
+                    stats_heir = y.groupby(X[col_heir]).agg(['count', 'mean'])
+                    smoove_heir = self._weighting(stats_heir['count'])
+                    scalar_heir = scalar * (1 - smoove_heir) + stats_heir['mean'] * smoove_heir
+                    scalar_heir_long = X[[col, col_heir]].drop_duplicates()
+                    scalar_heir_long.index = np.arange(1, scalar_heir_long.shape[0]+1)
+                    scalar = scalar_heir_long[col_heir].map(scalar_heir.to_dict())
+                    X.drop([col_heir], axis=1, inplace=True)
```

Comment:
we had a discussion in this project recently where a pandas guy discouraged using `inplace` and actually removed all (or at least most of) `inplace` usages from this repo. I don't remember the exact reason but it made sense to me and we should probably follow suit here as well. So just use  
`X = X.drop(...)`

Reply:
No problem.  Though nicely this line has been dropped due to the cleaned up methodology :)